db/BaseService/index.js

```diff
@@ -0,0 +1,57 @@
+const assert = require("assert");
+const { knex } = require("../helpers");
+
+class BaseService {
+    hasRecord(object) {
+        assert.ok(typeof object === "object", "object<object> is required");
+
+        return knex(this.name)
+            .where(object)
+            .select("id")
+            .then((rows) => {
+                knex.destroy();
+
+                return Boolean(rows.length);
+            });
+    }
+
+    fetchRecordById(recordId) {
+        assert.ok(typeof recordId === "number", "recordId<number> is required");
+
+        return knex(this.name)
+            .where("id", recordId)
+            .select()
+            .first()
+            .then(row => new this.Model(row))
+            .then((model) => {
+                knex.destroy();
+
+                return model;
+            });
+    }
+
+    fetchRecordsBy(object) {
+        assert.ok(typeof object === "object", "object<object> is required");
+
+        return knex(this.name)
+            .where(object)
+            .then(rows => rows.map(row => new this.Model(row)))
+            .then((models) => {
+                knex.destroy();
+
+                return models;
+            });
+    }
+
+    size() {
+        return knex(this.name)
+            .count("id as SIZE")
+            .then((total) => {
+                knex.destroy();
```

Comment:
why destroy here?

Reply:
Should the connection remain open indefinitely?

Reply:
knex maintains a connection pool. after the query, the connection is returned to the pool